### ISSUE TITLE
Ask for GH org too

### ISSUE THREAD

**Issue by gaborcsardi** [metacran/mason.github#5]
From https://github.com/metacran/mason.rpkg/issues/1

Related to 22e4cf2387683570c5f0d68b2ec9fb6d7e93739e, it might be worth doing this as:

GitHub username:

then

GitHub account to store the repository (with default being username)

I often create repositories that are not on my account but some lines (well, really just the copyright one at the bottom of the README) need the username, and others want the place where the package will reside.

**Comment by gaborcsardi** [comment]
I wonder if it is enough to ask for the org/user (to store at), and then if it turns out to be an org, then deduce the gh username automagically.

**Comment by gaborcsardi** [comment]
My intention is to automate as much as possible, and reduce the number of questions...

**Comment by richfitz** [comment]
Yeah, totally understand that.  Though it seems that questions that just require "Enter" with sensible defaults are pretty low pain.  Given how little the username is used if the repo is stored at an org that makes sense.

One thing to consider is how the question flow works when there is no internet though.

**Comment by gaborcsardi** [comment]
Yeah, I am testing without net connection occasionally. 
My current way is that whenever there is no net to deduce sg automatically, just ask.

**Comment by gaborcsardi** [comment]
Actually, I think you original suggestion makes sense. E.g.

```
? GitHub username / organization to store at (gaborcsardi): metacran
? GitHub username (gaborcsardi):
```

and the second is only asked if the answer to the first is an org. And when there is no network connection.

**Comment by richfitz** [comment]
Nice.  If you want to simulate loss of network during tests, check out [toxiproxy](http://toxiproxy.io) (potentially via [toxiproxyr](https://github.com/richfitz/toxiproxyr) [with apologies for self promotion]).

**Comment by gaborcsardi** [comment]
Thanks, I am one of the stargazers on toxyproxyr, and I think it is great. I have wanted something like that for a long time, to test metacran stuff, my pingr package, etc.

**Comment by gaborcsardi** [comment]
But actually I am far from writing real test cases, because I cannot simulate user input yet....

**Comment by gaborcsardi** [comment]
I am considering wrapping up `expect` in an R package, to test things like `mason`, but it is far from trivial.

**Comment by richfitz** [comment]
Oh, yeah.  Writing a thing to simulate sending input is going to be much harder than writing the package :)

**Comment by gaborcsardi** [comment]
You know https://en.wikipedia.org/wiki/Expect ?
With that it should be easier. I used it to automate ssh to some ridiculously secure servers that insist on me typing in a password.....

**Comment by richfitz** [comment]
No, I didn't!  Sounds fun (and presumably with tcltk baked into R you don't need anything external to make it work?).

Plus you'd get to write a testthat test `expect_expect`
